ignite/services/network/join_test.go

```diff
@@ -0,0 +1,544 @@
+		)
+		require.Error(t, joinErr)
+		require.Equal(t, expectedError, joinErr)
+		suite.AssertAllMocks(t)
+	})
+
+	t.Run("failed to send join request, failed to read node id", func(t *testing.T) {
+		var (
+			account        = testutil.NewTestAccount(t, testutil.TestAccountName)
+			suite, network = newSuite(account)
+			expectedError  = errors.New("failed to get node id")
+		)
+		suite.ChainMock.
+			On("NodeID", mock.Anything).
+			Return("", expectedError).
+			Once()
+
+		joinErr := network.Join(context.Background(), suite.ChainMock, testutil.LaunchID, WithPublicAddress(testutil.TCPAddress))
+		require.Error(t, joinErr)
+		require.Equal(t, expectedError, joinErr)
+		suite.AssertAllMocks(t)
+	})
+
+	t.Run("failed to send join request, failed to read default gentx", func(t *testing.T) {
+		var (
+			account        = testutil.NewTestAccount(t, testutil.TestAccountName)
+			suite, network = newSuite(account)
+			expectedError  = errors.New("failed to get default gentx path")
+		)
+
+		suite.ChainMock.On("NodeID", context.Background()).Return(testutil.NodeID, nil).Once()
+		suite.ChainMock.
+			On("DefaultGentxPath").
+			Return("", expectedError).
+			Once()
+
+		joinErr := network.Join(context.Background(), suite.ChainMock, testutil.LaunchID, WithPublicAddress(testutil.TCPAddress))
+		require.Error(t, joinErr)
+		require.Equal(t, expectedError, joinErr)
+		suite.AssertAllMocks(t)
+	})
+
+	t.Run("failed to send join request, failed to read genesis", func(t *testing.T) {
+		var (
+			account = testutil.NewTestAccount(t, testutil.TestAccountName)
+			tmp     = t.TempDir()
+			gentx   = testutil.NewGentx(
+				account.Address(networktypes.SPN),
+				TestDenom,
+				TestAmountString,
+				"",
+				testutil.PeerAddress,
+			)
+			gentxPath      = gentx.SaveTo(t, tmp)
+			suite, network = newSuite(account)
+			expectedError  = errors.New("failed to get genesis path")
+		)
+
+		suite.ChainMock.On("NodeID", context.Background()).Return(testutil.NodeID, nil).Once()
+		suite.ChainMock.On("DefaultGentxPath").Return(gentxPath, nil).Once()
+		suite.ChainMock.
+			On("GenesisPath").
+			Return("", expectedError).
+			Once()
+
+		joinErr := network.Join(context.Background(), suite.ChainMock, testutil.LaunchID, WithPublicAddress(testutil.TCPAddress))
+		require.Error(t, joinErr)
+		require.Equal(t, expectedError, joinErr)
+		suite.AssertAllMocks(t)
+	})
+
+	t.Run("failed to send join request, failed to read custom genesis", func(t *testing.T) {
+		var (
+			account        = testutil.NewTestAccount(t, testutil.TestAccountName)
+			gentxPath      = "invalid/path"
+			suite, network = newSuite(account)
+			expectedError  = errors.New("chain home folder is not initialized yet: invalid/path")
+		)
+
+		joinErr := network.Join(context.Background(), suite.ChainMock, testutil.LaunchID, WithCustomGentxPath(gentxPath))
+		require.Error(t, joinErr)
+		require.Equal(t, expectedError, joinErr)
+		suite.ChainMock.AssertNumberOfCalls(t, "NodeID", 0)
```

Comment:
I beileve running `AssertAllMocks` will already make sure that these funcs called zero times since we haven't declared any `Mock.On()...` above. Can we please check if that's the case and use `AssertAllMocks` to keep tests simple?